include/uuid/uuid.h

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2023 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef _UUID_H_
+#define _UUID_H_
+
+#include <algorithm>
+#include <bitset>
+#include <chrono>
+#include <cstddef>
+#include <ios>
+#include <iostream>
+#include <memory>
+#include <random>
+#include <sstream>
+#include <string>
+
+#include "random_gen.h"
+
+
+namespace uprotocol::uuid {
+
+class UUID {
+
+    public:
+
+        /**
+        * Constructor to build UUID object - the UUID generated internally
+        */
+        UUID();
+
+        /**
+        * Constructor to build UUID object - the UUID generated from input string
+        */
+        UUID(std::string &uuidStr);
+
+        /**
+        * Constructor to build UUID object - the UUID generated from byte array
+        *
+        */
+        UUID(uint8_t* bytes);
```

Comment:
We said that these constructors are from the deserializers.

Reply:
Refactored.